rust/kernel/sync/locked_by.rs

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A wrapper for data protected by a lock that does not wrap it.
```

Comment:
This looks like a nice idea.

A few thoughts:
  - I assume you are using this in your driver?
  - Is there any similar concept in the `std` we could reference?